heartbeat/monitors/active/http/respbody.go

```diff
@@ -28,30 +28,26 @@ import (
 
 	"github.com/elastic/beats/v7/heartbeat/reason"
 	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/mime"
 )
 
 // maxBufferBodyBytes sets a hard limit on how much we're willing to buffer for any reason internally.
 // since we must buffer the whole body for body validators this is effectively a cap on that.
 // 100MiB out to be enough for everybody.
 const maxBufferBodyBytes = 100 * units.MiB
 
-func processBody(resp *http.Response, config responseConfig, validator multiValidator) (common.MapStr, reason.Reason) {
+func processBody(resp *http.Response, config responseConfig, validator multiValidator) (common.MapStr, string, reason.Reason) {
 	// Determine how much of the body to actually buffer in memory
-	var bufferBodyBytes int
-	if validator.wantsBody() {
-		bufferBodyBytes = maxBufferBodyBytes
-	} else if config.IncludeBody == "always" || config.IncludeBody == "on_error" {
+	bufferBodyBytes := maxBufferBodyBytes
+	if !validator.wantsBody() && (config.IncludeBody == "always" || config.IncludeBody == "on_error") {
 		// If the user has asked for bodies to be recorded we only need to buffer that much
 		bufferBodyBytes = config.IncludeBodyMaxBytes
```

Comment:
I think we should set a lower bound on this value to be something that works with the mime type validator, maybe as low as the 128 bytes or similar? I'm just thinking of the case where a user sets it to a low value like 2.